server/webapp/WEB-INF/rails.new/lib/analytics_support.rb

```diff
@@ -0,0 +1,29 @@
+##########################################################################
+# Copyright 2017 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module AnalyticsSupport
+  include JavaImports
+
+  def with_pipeline_analytics_support(&block)
+    return unless block_given?
+
+    default_plugin_info_finder.allPluginInfos(PluginConstants.ANALYTICS_EXTENSION).each do |plugin|
```

Comment:
Wondering if this method should be moved to helper. 

You could as well use  `AnalyticsMetadataStore#allPluginInfos()` to go over analytics plugin_infos over using the `default_plugin_info_finder`.

Reply:
Yes. Should be a helper, I guess.

They talked about it. It's marked deprecated, and it looked like default_plugin_info_finder was starting to be a place to get info about plugins. No? We thought you would know best. So, you decide.

Reply:
I think in this case we should use the `AnalyticsMetadataStore` since we are dealing only with the Analytics Plugin.

Reply:
@maheshp @arvindsv The analytics_support file gets included in the application helper. Me and @marques-work figured that the analytics helper methods should be accessible from the everywhere. We created a module to include just for organizational reasons. 

That being said, we can totally move the methods into the application_helper directly if y'all would prefer

Reply:
@maheshp Never mind. Me and @arvindsv talked about it. I'm moving the method into the pipelines helper